### PR TITLE
#151991941 CZARs should be able to draw cards by clicking

### PR DESCRIPTION
**Pull request by tosmak16**
#### What does this PR do?
it allows czars to draw a black card before players can choose their answers

#### Description of Task to be completed?
* Add a function that manages draw card action
* Add a modal that pops up when the game starts and when a new round begins
* Add black cards that flip with animation effect when czar draws the black card

#### How should this be manually tested?
* Run this command "npm start".
* Click play as stranger
* Click the start button
* Click any of the black cards when the modal pops up
#### Any background context you want to provide?
none for now.

#### What are the relevant pivotal tracker stories?
#151991941

Screenshots
image
#### Questions:
none for now

### CODE REVIEW

File: config/socket/game.js [hunk 1]
```diff
@@ -323,9 +353,9 @@ class Game {
     });
   }
   /**
-   *
-   * @returns {void} description
-   * @param {any} cb
+   * @description this method triggers get all answer controller
+   * @returns {void}
+   * @param {function} cb
```
Comment (temilaj) [inline config/socket/game.js]: what does Cb mean?. Is there a better way to name this parameter?

File: config/socket/socket.js [hunk 2]
```diff
@@ -223,6 +223,23 @@ export default (io) => {
       }
     });
 
+    socket.on('drawCard', () => {
+      if (allGames[socket.gameID]) {
+        const thisGame = allGames[socket.gameID];
+        console.log('comparing', thisGame.players[0].socket.id, 'with', socket.id);
```
Comment (temilaj) [inline config/socket/socket.js]: Why do we have console log statements in source control?

File: public/js/controllers/game.js [hunk 3]
```diff
@@ -30,97 +28,94 @@ angular.module('mean.system')
...
-        return { cursor: 'pointer' };
+        return {
+          cursor: 'pointer'
+        };
       }
       return {};
     };
 
-    $scope.sendPickedCards = function () {
+    $scope.sendPickedCards = () => {
```
Comment (temilaj) [inline public/js/controllers/game.js:40]: Is there a reason for omitting documentation for other functions in your PR

**Review by temilaj — CHANGES_REQUESTED** [review]

File: public/js/controllers/game.js (hunk 3 above)
Comment (tosmak16) [reply public/js/controllers/game.js:40]: i didn't add those other functions i only wrote documentation for methods that i added to the codebase like drawCard method and setCzar method and also modified the other functions to arrow functions.

File: config/socket/socket.js (hunk 2 above)
Comment (tosmak16) [reply config/socket/socket.js]: I don't really understand why it is added in the codebase but i think it helps to know what socket emit and track the game state and changes. personally, i was able to understand how socket emit and listen to event. basically, it really helped me to understand the codebase. but if it is necessary to remove all the console log it is okay by me but i feel some of my team member would  need it for now.

File: config/socket/game.js (hunk 1 above)
Comment (tosmak16) [reply config/socket/game.js]: Well Cb is in the codebase before but i will make changes to it.

File: public/views/answers.html [hunk 4]
```diff
@@ -1,28 +1,81 @@
...
       </div>
     </div>
   </div>
+</div>
+
+
+
+<!-- space -->
 
+<!-- <div ng-cloak id="cards-container" ng-hide="game.state === 'game ended' || game.state ==='game dissolved'">
```
Comment (temilaj) [inline public/views/answers.html]: Is there any reason why this is comented out?

**Review by temilaj — CHANGES_REQUESTED** [review]
Made a comment. Please review

**Pull request merged**